crates/bevy_ecs/src/bundle.rs

```diff
@@ -1422,8 +1422,11 @@ impl Bundles {
             .or_insert_with(|| {
                 let (id, storages) =
                     initialize_dynamic_bundle(bundle_infos, components, Vec::from(component_ids));
-                self.dynamic_bundle_storages
-                    .insert_unique_unchecked(id, storages);
+                // SAFETY: We know the ID is unique
```

Comment:
I'm adding this comment to appease the lint, but I actually don't know if we do know if the ID is unique.

Reply:
I´ve checked the places where `Bundles::init_dynamic_info` was being used and most of them are internal facing, but I found `EntityWorldMut::insert_by_ids`, `EntityWorldMut::retain` and `EntityWorldMut::clear` which might be public facing I think?

`HashMap::insert_unique_unchecked` says:

```rust
    /// This operation is safe if a key does not exist in the map.
    ///
    /// However, if a key exists in the map already, the behavior is unspecified:
    /// this operation may panic, loop forever, or any following operation with the map
    /// may panic, loop forever or return arbitrary result.
```

So I'm unsure if we should say something like the caller ensures the uniqueness, and even add some notes in the `EntityWorldMut` methods if applicable? I guess I'm thinking out loud. 😅

Reply:
So, I'm honestly fine just changing this to `insert` and leaving it at that :p

Reply:
Okay, so, I double-checked and the bundle ID *is* always increasing here, and without concurrency, we don't need to worry about data races. So, I updated the comment to clarify it is actually unique and leaving it.